### PR TITLE
nss-systemd: also expose shadow/gshadow entries from userdb records

### PR DESCRIPTION
**Pull request by poettering**
Split out of #19528

### CODE REVIEW

File: src/nss-systemd/nss-systemd.c [hunk 1]
```diff
@@ -355,6 +505,46 @@ enum nss_status _nss_systemd_setgrent(int stayopen) {
...
+        int r;
+
+        PROTECT_ERRNO;
+        NSS_ENTRYPOINT_BEGIN;
+
+        if (_nss_systemd_is_blocked())
+                return NSS_STATUS_NOTFOUND;
+
+        _cleanup_(pthread_mutex_unlock_assertp) pthread_mutex_t *_l =
+                pthread_mutex_lock_assert(&getspent_data.mutex);
```
Comment (yuwata) [inline src/nss-systemd/nss-systemd.c]: Please add `(void) _l;`.

**Review by yuwata — APPROVED** [review]
LGTM, but CIs are unhappy.

File: src/nss-systemd/userdb-glue.c [hunk 2]
```diff
@@ -129,6 +129,85 @@ enum nss_status userdb_getpwuid(
...
+                char *buffer,
+                size_t buflen) {
+
+        const char *hashed;
+        size_t required;
+
+        assert(hr);
+        assert(spwd);
+
+        assert_se(hr->user_name);
```
Comment (yuwata) [inline src/nss-systemd/userdb-glue.c]: Why `assert_se()` instead of `assert()`?

Comment (poettering) [reply src/nss-systemd/userdb-glue.c]: no reason besides copy/paste sloppiness ;-)

**Comment by poettering** [comment]
Force pushed a new version. Only changes are the suggested one. Upgrading label

**Pull request merged**